src/test/java/it/org/jfrog/bamboo/utils/Utils.java

```diff
@@ -0,0 +1,97 @@
+    public static final String BAMBOO_TEST_URL = "http://localhost:6990/bamboo";
+
+    // Environment variables names
+    public static final String GRADLE_HOME_ENV = "GRADLE_HOME";
+    public static final String MAVEN_HOME_ENV = "MAVEN_HOME";
+
+    // Environment variables values
+    public static final String GRADLE_HOME = System.getenv(GRADLE_HOME_ENV);
+    public static final String MAVEN_HOME = System.getenv(MAVEN_HOME_ENV);
+
+    /**
+     * Create overriding plan variables. The Bamboo Artifactory plugin uses them in order to override
+     * configured Artifactory URL, username, password and repositories.
+     *
+     * @param localRepoKey - Local test repository key
+     * @param jcenter      - JCenter repository key
+     * @return overriding plan variables
+     */
+    public static Map<String, String> createOverrideVars(String localRepoKey, String jcenter) {
+        return new HashMap<String, String>() {{
+            put(OVERRIDE_ARTIFACTORY_DEPLOYER_URL, ARTIFACTORY_URL);
+            put(OVERRIDE_ARTIFACTORY_DEPLOYER_USERNAME, ARTIFACTORY_USERNAME);
+            put(OVERRIDE_ARTIFACTORY_DEPLOYER_PASSWORD, ARTIFACTORY_PASSWORD);
+            put(OVERRIDE_ARTIFACTORY_DEPLOY_REPO, localRepoKey);
+            put(OVERRIDE_ARTIFACTORY_RESOLVER_URL, ARTIFACTORY_URL);
+            put(OVERRIDE_ARTIFACTORY_RESOLVER_USERNAME, ARTIFACTORY_USERNAME);
+            put(OVERRIDE_ARTIFACTORY_RESOLVER_PASSWORD, ARTIFACTORY_PASSWORD);
+            put(OVERRIDE_ARTIFACTORY_RESOLVE_REPO, jcenter);
+        }};
+    }
+
+    /**
+     * Create common plan environment variables.
+     *
+     * @return plan environment variables
+     */
+    public static Map<String, String> createEnv() {
+        return new HashMap<String, String>() {{
+            // The collect env methodology should ignore this variable:
+            put("DONT_COLLECT", "FOO");
+
+            // The collect env methodology should add this variable:
+            put("COLLECT", "BAR");
+        }};
+    }
+
+    /**
+     * Download the build log after test failure.
+     *
+     * @param log         - The tests logger
+     * @param buildNumber - The build number of the test plan
+     * @param planKey     - The test plan key
+     * @return the build log or empty string if error occurred.
+     */
+    public static String downloadBuildLog(Logger log, int buildNumber, String planKey) {
+        String url = String.format("%s/download/%s-JOB1/build_logs/%s-JOB1-%s.log", BAMBOO_TEST_URL, planKey, planKey, buildNumber);
+        HttpGet request = new HttpGet(url);
+        request.setHeader(HttpHeaders.AUTHORIZATION, ENCODED_AUTH);
+        try (CloseableHttpClient httpClient = HttpClients.createDefault();
+             CloseableHttpResponse response = httpClient.execute(request)) {
+            HttpEntity entity = response.getEntity();
+            if (entity != null) {
+                return EntityUtils.toString(entity);
```

Comment:
I think the enrity should be consumed before returning:
```
EntityUtils.consumeQuietly(response.getEntity());
```

Reply:
The entity's Inputstream can be read only once, therefore it should be closed in the toString method. See:
https://github.com/apache/httpcomponents-core/blob/rel/v4.4.5/httpcore/src/main/java/org/apache/http/util/EntityUtils.java#L252